internal/advertisement-operator/config-watcher.go

```diff
@@ -4,91 +4,65 @@ import (
 	"context"
 	protocolv1 "github.com/liqoTech/liqo/api/advertisement-operator/v1"
 	policyv1 "github.com/liqoTech/liqo/api/cluster-config/v1"
+	"github.com/liqoTech/liqo/pkg/clusterConfig"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/watch"
 	"k8s.io/klog"
 )
 
-func (b *AdvertisementBroadcaster) WatchConfiguration(kubeconfigPath string) error {
-	configClient, err := policyv1.CreateClusterConfigClient(kubeconfigPath)
```

Comment:
Why is it necessary to change this and do not exit if watch is broken?

Reply:
The change allow us to have one watcher function and pass it only a function to handle our specific need in the component

For the exit I can add a logging and a os exit instead of return

Reply:
Ok, got it. I think that we can consider a standard policy is: if it is unable to get the configuration, the process will fail.

Reply:
ok, and I think it's implemented [here](https://github.com/LiqoTech/liqo/blob/aol/cluster-config-improvement/pkg/clusterConfig/watchConfig.go#L33)